panel/io/handlers.py

```diff
@@ -180,11 +180,182 @@ def capture_code_cell(cell):
 """)
     return code
 
+def autoreload_handle_exception(handler, module, e):
+    if not config.autoreload:
+        handle_exception(handler, e)
+        return
+
+    from ..pane import Alert
+
+    # Clean up module
+    del sys.modules[module.__name__]
+    state.curdoc.modules._modules.remove(module)
+
+    # Serve error
+    e_msg = str(e).replace('\033[1m', '<b>').replace('\033[0m', '</b>')
+    tb = html.escape(traceback.format_exc()).replace('\033[1m', '<b>').replace('\033[0m', '</b>')
+    Alert(
+        f'<b>{type(e).__name__}</b>: {e_msg}\n<pre style="overflow-y: auto">{tb}</pre>',
+        alert_type='danger', margin=5, sizing_mode='stretch_width'
+    ).servable()
+
+def run_app(handler, module, doc, post_run=None):
+    try:
+        old_doc = curdoc()
+    except RuntimeError:
+        old_doc = None
+        bk_set_curdoc(doc)
+
+    sessions = []
+
+    def post_check():
+        newdoc = curdoc()
+        # Do not let curdoc track modules when autoreload is enabled
+        # otherwise it will erroneously complain that there is
+        # a memory leak
+        if config.autoreload:
+            newdoc.modules._modules = []
+
+        # script is supposed to edit the doc not replace it
+        if newdoc is not doc:
+            raise RuntimeError("%s at '%s' replaced the output document" % (handler._origin, handler._runner.path))
+
+    try:
+        state._launching.append(doc)
+        with _monkeypatch_io(handler._loggers):
+            with patch_curdoc(doc):
+                with profile_ctx(config.profiler) as sessions:
+                    with record_modules(handler=handler):
+                        handler._runner.run(module, post_check)
+                        if post_run:
+                            post_run()
```

Comment:
```suggestion
        with (
            _monkeypatch_io(handler._loggers),
            patch_curdoc(doc),
            profile_ctx(config.profiler) as sessions,
            record_modules(handler=handler),
        ):
            handler._runner.run(module, post_check)
            if post_run:
                post_run()
```

First supported in Python 3.10. I'm unsure if this will work, e.g, a context manager depends on a setting in the previous one, but I already did the code changes and didn't want to delete it. 

If we want it now we could do something like this:

``` python
with context1 as a,\
     context2 as b:
    pass
``

Reply:
Oooh, neat, but we still support 3.9 😢